### PR TITLE
Crate reorganization

### LINKED ISSUE #320: Repository reorganization
Proposition:

- Move everything related to the `gl` crate (`Cargo.toml`, `src/gl`, `tests` and `examples`) in a `gl` subdirectory.
- Move `gl_generator`, `khronos_api` and `gl_common` directories from `src` to the crate root.
- Add a `gl_tests` crate (which we won't publish) that is used to test the output of `gl_generator` instead of `gl` itself (solves #314).

All five crates (`gl`, `gl_generator`, `gl_common`, `khronos_api`, `gl_tests`) would be at the same level at the root of the repository, alongside with `.gitignore`, `.travis.yml`, etc.

### PR DESCRIPTION
**Pull request by tomaka**
Close #320

[Preview](https://github.com/tomaka/gl-rs/tree/reorganization)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,8 +6,13 @@ env:
   global:
     - secure: SH7xrrH8ONXcrpHzKVJsBuLX1AikKBTBBI0SsEfCWfDZsAur0rQW5ZlPPorhQ0iZUbvw5xwY26T+0EYdBAWOwp7AOBM2iwoGZ15ERTd8o2JX3ZlMQ9AvDHNaXAc6YToj69DaiaimX+oX9DtZeRfQ309Lkp4D1YMrf4X/jhNpQdI=
 
+script:
+  - cargo test --verbose --manifest-path gl_generator/Cargo.toml
+  - cargo test --verbose --manifest-path gl/Cargo.toml
+  - cargo test --verbose --manifest-path gl_tests/Cargo.toml
```
Comment (bjz) [inline .travis.yml]: Could we include a `.sh` that does this?

Comment (tomaka) [inline .travis.yml]: Just for three lines? :huh:

Comment (bjz) [inline .travis.yml]: For ease of local testing.

**Comment by tomaka** [comment]
> $ ./test-all.sh
> /home/travis/build.sh: line 41: ./test-all.sh: Permission denied

Maybe a stupid question, but why does it do that?

**Comment by bjz** [comment]
Are the permissions ok? +x?

**Comment by tomaka** [comment]
I'm on windows.

**Pull request merged**